File: src/AppRegistryService.Contract/Responses/AppInstallerInfo.cs
namespace AppRegistryService.Contract.Responses;

/// <summary>
/// Defines an application release installer.
/// </summary>
public sealed class AppInstallerInfo
{
    /// <summary>
    /// Unique installer identifier.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Release identifier link.
    /// </summary>
    public Guid ReleaseId { get; set; }

    /// <summary>
    /// Installer file uri.
    /// </summary>
    public Uri? Uri { get; set; }

    /// <summary>
    /// Installer title.
    /// </summary>
    public string? Title { get; set; }

    /// <summary>
    /// Installer description.
    /// </summary>
    public string? Description { get; set; }

    /// <summary>
    /// Installer file size in bytes.
    /// </summary>
    public long? Size { get; set; }

    /// <summary>
    /// Installer additional (optional) files size in bytes.
    /// </summary>
    public long? AdditionalSize { get; set; }
}

File: web/src/responses/AppInstallerInfo.ts
/**
 * Defines an application release installer.
 */
export interface AppInstallerInfo {
	/**
	 * Unique installer identifier.
	 */
	id: string;

	/**
	 * Release identifier link.
	 */
	releaseId: string;

	/**
	 * Installer file uri.
	 */
	uri?: string;

	/**
	 * Installer title.
	 */
	title?: string;

	/**
	 * Installer description.
	 */
	description?: string;

	/**
	 * Installer file size in bytes.
	 */
	size?: number;

	/**
	 * Installer additional (optional) files size in bytes.
	 */
	additionalSize?: number;
}
